File: IOS_swift/WitSDK/Sensor/Modular/Processor/Protocol/IDataProcessor.swift
//
//  数据处理器协议
//
//  Created by huangyajun on 2022/8/31.
//

import Foundation

protocol IDataProcessor{
    
    // 打开设备时
    func onOpen(deviceModel:DeviceModel)
    
    // 关闭设备时
    func onClose()
    
    // 设备数据实时更新时
    func onUpdate(deviceModel:DeviceModel)
}

File: Android_Java/Old_Example/wit-example-ble5/app/src/main/java/com/wit/witsdk/sensor/modular/processor/interfaces/IDataProcessor.java
package com.wit.witsdk.sensor.modular.processor.interfaces;

import com.wit.witsdk.sensor.modular.device.DeviceModel;

/**
 * 数据处理器接口
 *
 * @author huangyajun
 * @date 2022/4/21 11:22
 */
public interface IDataProcessor {

    /**
     * 打开设备时
     *
     * @author huangyajun
     * @date 2022/4/21 11:24
     */
    void OnOpen(DeviceModel deviceModel);

    /**
     * 关闭传感器时
     *
     * @author huangyajun
     * @date 2022/4/21 11:24
     */
    void OnClose();

    /**
     * 设备实时数据更新时
     *
     * @author huangyajun
     * @date 2022/4/21 11:24
     */
    void OnUpdate(DeviceModel deviceModel);

}
